### DISCUSSION TITLE
v1beta1 + v0.10.0

### DISCUSSION THREAD

**Discussion by EmilienM** [kubernetes-sigs/cluster-api-provider-openstack#1844] [General]
A few of us are actively looking at releasing CAPO v1beta1 (with a v0.10.0 branch) in the following weeks.
We would like to ask our community what they want to be in there before we cut the release.

I've dressed a list of things that I think would be required, I plan to maintain that list while the discussion is ongoing:

- [x] https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1828 (@EmilienM)
- [x] Very likely https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1835 (@EmilienM)
- [x] https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1856 (@dulek)
- [x] https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1826 (@EmilienM)
- [x] https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1878 (@EmilienM)
- [x] https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1903 (@EmilienM)
- [ ] https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1850 (@mdbooth)

Please provide feedback and be involved!

**Comment by MaysaMacedo** [comment]
@EmilienM For the dual-stack support on NodeCIDR, we would need to create one additional subnet,  IPv6 one this time and add it to the router. So this wouldn't be a large change if the Subnet uses a fixed address mode like dhcpv6-stateful. If we would like to let the user choose the address mode, that would be an additional API and could expand scope of this work.

On another topic, have we considered the following work for beta1? https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1836

**Reply by EmilienM** [reply]
> @EmilienM For the dual-stack support on NodeCIDR, we would need to create one additional subnet, IPv6 one this time and add it to the router. So this wouldn't be a large change if the Subnet uses a fixed address mode like dhcpv6-stateful. If we would like to let the user choose the address mode, that would be an additional API and could expand scope of this work.

Let's not do that, they can always do BYON, I don't think we can / should support every use case.

> On another topic, have we considered the following work for beta1? #1836

Added. Thanks!

**Reply by mdbooth** [reply]
I don't think #1836 is a v1beta1 issue: I don't think fixing it can break any existing working configuration. Doesn't mean we shouldn't fix it, I just don't think it's a v1beta1 blocker.

**Comment by huxcrux** [comment]
Would be nice to have https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1762 in v1beta1

I will check if https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1687 still is a problem (hope to have it done today or tomorrow) to check if it's still a problem and in case if we can fixa a PR with a fix

**Reply by EmilienM** [reply]
Thanks for your input, I'll add it to the list but also wait for @mdbooth 's feedback on this one.
